Mathlib/CategoryTheory/Bicategory/NaturalTransformation/Strong.lean

```diff
@@ -0,0 +1,202 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Calle Sönne
+-/
+
+import Mathlib.CategoryTheory.Bicategory.Functor.Pseudofunctor
+import Mathlib.CategoryTheory.Bicategory.NaturalTransformation.Oplax
+
+/-!
+# Strong natural transformations
+
+A strong natural transformation is an oplax natural transformation such that each component 2-cell
+is an isomorphism.
+
+## Main definitions
+
+* `StrongNatTrans F G` : strong natural transformations between oplax functors `F` and `G`.
+* `mkOfOplax η η'` : given an oplax natural transformation `η` such that each component 2-cell
+  is an isomorphism, `mkOfOplax` gives the corresponding strong natural transformation.
+* `StrongNatTrans.vcomp η θ` : the vertical composition of strong natural transformations `η`
+  and `θ`.
+* `StrongNatTrans.category F G` : a category structure on pseudofunctors between `F` and `G`,
+  where the morphisms are strong natural transformations.
+
+## TODO
+
+After having defined lax functors, we should define 3 different types of strong natural
+transformations:
+* strong natural transformations between oplax functors (as defined here).
+* strong natural transformations between lax functors.
+* strong natural transformations between pseudofunctors. From these types of strong natural
+  transformations, we can define the underlying natural transformations between the underlying
+  oplax resp. lax functors. Many properties can then be inferred from these.
+
+## References
+* [Niles Johnson, Donald Yau, *2-Dimensional Categories*](https://arxiv.org/abs/2002.06055)
+
+-/
+
+namespace CategoryTheory
+
+open Category Bicategory
+
+open scoped Bicategory
+
+universe w₁ w₂ v₁ v₂ u₁ u₂
+
+variable {B : Type u₁} [Bicategory.{w₁, v₁} B] {C : Type u₂} [Bicategory.{w₂, v₂} C]
+
+/-- A strong natural transformation between oplax functors `F` and `G` is a natural transformation
+that is "natural up to 2-isomorphisms".
+
+More precisely, it consists of the following:
+* a 1-morphism `η.app a : F.obj a ⟶ G.obj a` for each object `a : B`.
+* a 2-isomorphism `η.naturality f : F.map f ≫ app b ⟶ app a ≫ G.map f` for each 1-morphism
+`f : a ⟶ b`.
+* These 2-isomorphisms satisfy the naturality condition, and preserve the identities and the
+compositions modulo some adjustments of domains and codomains of 2-morphisms.
+-/
+structure StrongNatTrans (F G : OplaxFunctor B C) where
```

Comment:
```suggestion
structure StrongOplaxNatTrans (F G : OplaxFunctor B C) where
```
The name has be changed, otherwise there will be a problem with lax/pseudo-functors.

Reply:
Very good point, thanks!